src/hooks/useAsync.ts

```diff
@@ -25,11 +29,13 @@ const useAsync = <T>(asyncCall: () => Promise<T>, dependencies: unknown[], clear
         isCurrent && setError(err)
       })
       .finally(() => {
+        clearTimeout(loadingTimeout)
         isCurrent && setLoading(false)
```

Comment:
Just noticed that we call `setLoading` twice here

Reply:
Where exactly? It's only called before the promise call and in `finally`.

Reply:
In `finally` it is called via `loadingTimeout` which sets it to true and then in the next line it is set to false but it should only be set to false once (in a promisified way) if I understand it correctly.